app/models/spree/product_property.rb

```diff
@@ -2,12 +2,9 @@
 
 module Spree
   class ProductProperty < ApplicationRecord
-    self.belongs_to_required_by_default = false
-
-    belongs_to :product, class_name: "Spree::Product", touch: true
+    belongs_to :product, class_name: "Spree::Product", touch: true, optional: true
```

Comment:
It don't think it makes sense to have the product optional here. 
I checked the production server and none of them have a product property with a `product_id` set to null so it's safe to the change. @openfoodfoundation/core-devs  what do you think ?

Reply:
Well if there are no null `product_id`, I guess I could also update the migration file.

Reply:
Indeed, ProductProperty is a join table between Product and Property, so the association is definitely required.

So both the product_id and property_id columns can become `NOT NULL`. I'm not sure if Gaetan checked all prod servers yet, so we might need to check that before deploying.

Reply:
Cool. So I have made the update.